### PR TITLE
[SPARK-19949][SQL][follow-up] move FailureSafeParser from catalyst to sql core

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

The `FailureSafeParser` is only used in sql core, it doesn't make sense to put it in catalyst module.

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @gatorsmile

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 1]
```diff
@@ -27,11 +27,10 @@ import org.apache.spark.Partition
 import org.apache.spark.annotation.InterfaceStability
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.catalyst.json.{CreateJacksonParser, JacksonParser, JSONOptions}
-import org.apache.spark.sql.catalyst.util.FailureSafeParser
 import org.apache.spark.sql.execution.LogicalRDD
 import org.apache.spark.sql.execution.command.DDLUtils
 import org.apache.spark.sql.execution.datasources.csv._
-import org.apache.spark.sql.execution.datasources.DataSource
+import org.apache.spark.sql.execution.datasources.{DataSource, FailureSafeParser}
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]:
```Scala
import org.apache.spark.sql.execution.datasources.{DataSource, FailureSafeParser}
import org.apache.spark.sql.execution.datasources.csv._
```

**Comment by gatorsmile** [comment]
LGTM pending Jenkins

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**